File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/TicketNewResponse.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Response returned after creating a new support ticket.
 */
@Schema(description = "Response returned after creating a new support ticket.")
@Validated
@NotUndefined



public class TicketNewResponse   {
  @JsonProperty("ticket_id")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private Integer ticketId = null;

  @JsonProperty("text")

  private String text = null;

  @JsonProperty("success")

  private Boolean success = null;


  public TicketNewResponse ticketId(Integer ticketId) { 

    this.ticketId = ticketId;
    return this;
  }

  /**
   * Get ticketId
   * @return ticketId
   **/
  
  @Schema(description = "")
  
  public Integer getTicketId() {  
    return ticketId;
  }



  public void setTicketId(Integer ticketId) { 
    this.ticketId = ticketId;
  }

  public TicketNewResponse text(String text) { 

    this.text = text;
    return this;
  }

  /**
   * Get text
   * @return text
   **/
  
  @Schema(required = true, description = "")
  
  @NotNull
  public String getText() {  
    return text;
  }



  public void setText(String text) { 

    this.text = text;
  }

  public TicketNewResponse success(Boolean success) { 

    this.success = success;
    return this;
  }

  /**
   * Get success
   * @return success
   **/
  
  @Schema(required = true, description = "")
  
  @NotNull
  public Boolean isSuccess() {  
    return success;
  }



  public void setSuccess(Boolean success) { 

    this.success = success;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    TicketNewResponse ticketNewResponse = (TicketNewResponse) o;
    return Objects.equals(this.ticketId, ticketNewResponse.ticketId) &&
        Objects.equals(this.text, ticketNewResponse.text) &&
        Objects.equals(this.success, ticketNewResponse.success);
  }

  @Override
  public int hashCode() {
    return Objects.hash(ticketId, text, success);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class TicketNewResponse {\n");
    
    sb.append("    ticketId: ").append(toIndentedString(ticketId)).append("\n");
    sb.append("    text: ").append(toIndentedString(text)).append("\n");
    sb.append("    success: ").append(toIndentedString(success)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/TicketNewResponse.cpp


#include "TicketNewResponse.h"

using namespace Tiny;

TicketNewResponse::TicketNewResponse()
{
	text = std::string();
	success = bool(false);
	ticket_id = int(0);
}

TicketNewResponse::TicketNewResponse(std::string jsonString)
{
	this->fromJson(jsonString);
}

TicketNewResponse::~TicketNewResponse()
{

}

void
TicketNewResponse::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *textKey = "text";

    if(object.has_key(textKey))
    {
        bourne::json value = object[textKey];



        jsonToValue(&text, value, "std::string");


    }

    const char *successKey = "success";

    if(object.has_key(successKey))
    {
        bourne::json value = object[successKey];



        jsonToValue(&success, value, "bool");


    }

    const char *ticket_idKey = "ticket_id";

    if(object.has_key(ticket_idKey))
    {
        bourne::json value = object[ticket_idKey];



        jsonToValue(&ticket_id, value, "int");


    }


}

bourne::json
TicketNewResponse::toJson()
{
    bourne::json object = bourne::json::object();





    object["text"] = getText();






    object["success"] = isSuccess();






    object["ticket_id"] = getTicketId();



    return object;

}

std::string
TicketNewResponse::getText()
{
	return text;
}

void
TicketNewResponse::setText(std::string text)
{
	this->text = text;
}

bool
TicketNewResponse::isSuccess()
{
	return success;
}

void
TicketNewResponse::setSuccess(bool success)
{
	this->success = success;
}

int
TicketNewResponse::getTicketId()
{
	return ticket_id;
}

void
TicketNewResponse::setTicketId(int ticket_id)
{
	this->ticket_id = ticket_id;
}



